src/documents/signals/handlers.py

```diff
@@ -499,3 +501,25 @@ def add_to_index(sender, document, **kwargs):
     from documents import index
 
     index.add_or_update_document(document)
+
+
+@receiver(django_q.signals.pre_enqueue)
+def init_paperless_task(sender, task, **kwargs):
+    if task["func"] == "documents.tasks.consume_file":
+        paperless_task, created = PaperlessTask.objects.get_or_create(
+            task_id=task["id"],
+        )
+        paperless_task.name = task["name"]
+        paperless_task.created = task["started"]
+        paperless_task.save()
+
+
+@receiver(models.signals.post_save, sender=django_q.tasks.Task)
```

Comment:
Ah cool yea that makes more sense but weirdly I cant get this loaded. Official docs show v1.3.6 which doesnt have this signal ( https://django-q.readthedocs.io/en/latest/signals.html ) but then this repo https://github.com/Koed00/django-q/blob/master/docs/signals.rst shows it but when I tried installing 1.3.9 I still dont see it (i.e. just get no attribute `post_execute`). Huh?

<img width="264" alt="Screen Shot 2022-06-06 at 12 15 27 AM" src="https://user-images.githubusercontent.com/4887959/172113656-365ca814-f419-4651-b7b7-9378b8b88b5b.png">

Reply:
Ah, I see the issue.  It's only in the `master` and never got released.  So I think the `post_save` is the best possible now, until #1014 goes in.

Reply:
So in actually looking at this after rebasing on dev I think `post_save` is still more helpful to use because that gives us the actual `Task` model in the database as a `OneToOneField` which allows us quicker access to all the data that object already holds, like `paperless_task.attempted_task.success` etc

What do you think?